### PR TITLE
Make it easier to load/mock clients/libs on the Clients class

### PR DESCRIPTION
**Pull request by arsalansufi**
This PR makes it easier to load/mock clients/libs on the Clients class, while also maintaining good typing. It's a port of https://github.com/Clever/family-portal/pull/22. See that PR for more context!

### CODE REVIEW

File: src/server/config.ts [hunk 1]
```diff
@@ -1 +1,2 @@
+export const IS_TEST = Boolean(process.env.TEST);
 export const PORT = process.env.PORT || 5020;
```
Comment (arsalansufi) [inline src/server/config.ts:2]: Improvements to this config file are coming in a separate PR

File: src/server/lib/Clients.ts [hunk 2]
```diff
@@ -1,52 +1,59 @@
...
+type ClientsMap = Omit<typeof ClientDefinitions, "prototype">;
 
-  static get spongeBobService() {
-    return this._spongeBobService;
-  }
-}
+/**
+ * An object where all clients can optionally be present and all methods of a client are optional.
+ * Ideal for passing in mocks of clients.
+ */
```
Comment (prime-time) [inline src/server/lib/Clients.ts:56]: 👌

**Review by prime-time — APPROVED** [review]
Looks good to me

**Pull request merged**